src/main/scala/com/lucidworks/spark/PartitionByTimeFeature.scala

```diff
@@ -0,0 +1,85 @@
+package com.lucidworks.spark
+
+import java.text.SimpleDateFormat
+import java.util.{Date, TimeZone}
+import java.util.concurrent.TimeUnit
+import java.util.regex.{Matcher, Pattern}
+
+import com.lucidworks.spark.util.QueryConstants._
+import com.lucidworks.spark.util.{SolrQuerySupport, SolrSupport}
+import org.apache.solr.client.solrj.SolrQuery
+import org.apache.spark.Logging
+
+/**
+  * Created by akashmehta on 6/17/16.
+  */
+class PartitionByTimeFeature(val conf:SolrConf) extends Logging {
+  private val TIME_PERIOD_PATTERN: Pattern = Pattern.compile("^(\\d{1,4})(MINUTES|HOURS|DAYS)$")
```

Comment:
consider renaming this class as "Feature" is a Fusion-specific concept. This is simply config